bukuserver/server.py

```diff
@@ -80,6 +81,22 @@ def init_locale(app, context_processor=lambda: {}):
         app.context_processor(lambda: {'lang': '', **context_processor()})
 
 
+# handling popup= URL argument
+def before_request():
+    _post_popup = request.headers.get('Content-Type') != 'application/json' and request.data.get('popup')
+    flask.g.popup = request.args.get('popup') or _post_popup
+
+# applying popup= to the redirect URL
+def after_request(response):
+    if flask.g.popup and 'Location' in response.headers:
+        _scheme, _netloc, _path, _query, _fragment = urlsplit(response.headers['Location'])
+        _params = parse_qs(_query)
+        if not _params.get('popup'):
+            _query = '&'.join(s for s in [_query, 'popup=True'] if s)
+            response.headers['Location'] = urlunsplit((_scheme, _netloc, _path, _query, _fragment))
+    return response
```

Comment:
Attaching `popup=True` to redirect URLs, if necessary.

This appears to be the only sane way to do so, since there are no hooks to override behaviour of "Save and Add Another"/"Save and Continue Editing" buttons.